include/fiction/algorithms/simulation/sidb/displacement_robustness_domain.hpp

```diff
@@ -0,0 +1,670 @@
+    /**
+     * Parameters to check the operation status of the SiDB layout.
+     */
+    is_operational_params operational_params{};
+    /**
+     * SiDBs in the given layout which shall not be affected by variations.
+     */
+    std::set<CellType> fixed_sidbs{};
+    /**
+     * This flag controls whether the displacement in the y-direction can lead to changes in the Si dimer.
+     */
+    dimer_displacement_policy dimer_policy = dimer_displacement_policy::STAY_ON_ORIGINAL_DIMER;
+};
+
+/**
+ * Statistics for the displacement robustness domain computation.
+ */
+struct displacement_robustness_domain_stats
+{
+    /**
+     * Total runtime in seconds to determine the robustness of the passed SiDB layout.
+     */
+    mockturtle::stopwatch<>::duration time_total{0};
+    /**
+     * The number of operational SiDB layouts resulting from the given layout by displacements.
+     */
+    std::size_t num_operational_sidb_displacements{0};
+    /**
+     * The number of non-operational SiDB layouts resulting from the given layout by displacements.
+     */
+    std::size_t num_non_operational_sidb_displacements{0};
+};
+
+namespace detail
+{
+
+template <typename Lyt, typename TT>
+class displacement_robustness_domain_impl
+{
+  public:
+    /**
+     * Standard constructor. Initializes the layout, the truth table, the parameters, and the statistics.
+     *
+     * @param lyt SiDB cell-level layout to be evaluated regarding displacement robustness.
+     * @param spec Expected Boolean function of the layout given as a multi-output truth table.
+     * @param ps Parameters for the displacement robustness computation.
+     * @param st Statistics related to the displacement robustness computation.
+     */
+    displacement_robustness_domain_impl(const Lyt& lyt, const std::vector<TT>& spec,
+                                        const displacement_robustness_domain_params<cell<Lyt>>& ps,
+                                        displacement_robustness_domain_stats&                   st) noexcept :
+            layout{lyt},
+            params{ps},
+            stats{st},
+            truth_table{spec},
+            generator(rd())
+    {
+        assert(
+            (is_operational(layout, truth_table, params.operational_params).first == operational_status::OPERATIONAL) &&
+            "The given layout is not a valid SiDB layout for the given Boolean function");
+
+        if (params.analysis_mode ==
+            displacement_robustness_domain_params<cell<Lyt>>::displacement_analysis_mode::RANDOM)
+        {
+            assert(params.percentage_of_analyzed_displaced_layouts >= 0.0 &&
+                   params.percentage_of_analyzed_displaced_layouts <= 1.0 &&
+                   "percentage_of_analyzed_displaced_layouts must be between 0.0 and 1.0");
+        }
+
+        sidbs_of_the_original_layout.reserve(layout.num_cells());
+        layout.foreach_cell([&](const auto& c) { sidbs_of_the_original_layout.push_back(c); });
```

Comment:
```suggestion
        layout.foreach_cell([&sidbs_of_the_original_layout](const auto& c) { sidbs_of_the_original_layout.push_back(c); });
```

Reply:
``sidbs_of_the_original_layout`` is a member variable. Hence, I will choose ``this``.